Tasks/SqlAzureDacpacDeploymentV1/DeploySqlAzure.ps1

```diff
@@ -78,23 +77,29 @@ try {
         $sb = New-Object System.Data.Common.DbConnectionStringBuilder
         $sb.set_ConnectionString($connectionString.toLower())
 
-        if($sb['data source']) {
-          $serverName = $sb['data source']
-        } elseif ($sb['server']) {
-          $serverName = $sb['server']
+        if ($sb['data source']) {
+            $serverName = $sb['data source']
+        }
+        elseif ($sb['server']) {
+            $serverName = $sb['server']
         }
 
-        if($sb['initial catalog']) {
-          $databaseName = $sb['initial catalog']
-        } elseif ($sb['database']) {
-          $databaseName = $sb['database']
+        if ($sb['initial catalog']) {
+            $databaseName = $sb['initial catalog']
+        }
+        elseif ($sb['database']) {
+            $databaseName = $sb['database']
         }
 
         Write-Verbose "Retrieved SQL server name : $serverName  and database : $databaseName from connection string"
-    } elseif ($authenticationType -eq "aadAuthenticationPassword") {
+    }
+    elseif ($authenticationType -eq "aadAuthenticationPassword") {
         $sqlUsername = $aadSqlUserName;
         $sqlPassword = $aadSqlPassword;
     }
+    elseif ($authenticationType -eq "servicePrincipal") {
+        $accessToken = (Get-AzureRMAccessToken -endpoint $endpoint -overrideResourceType "https://database.windows.net/").access_token
```

Comment:
Instead of hardcoding `https://database.windows.net/`, get the value from `$endpoint.Data.sqlDatabaseDnsSuffix`. The output is in form `.database.windows.net`. You will have to remove dot(.) from the start and add https.

This is needed because the task also targets other national clouds, so this value will be different for different clouds.

Reply:
I will investigate all those points by the week-end. Need time to